File: 1_TwoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        i=0
        while i<len(nums):
            if i==len(nums)-1:
                return "No solutions"
            r=target-nums[i]
            nums_follow=nums[i+1:]
            if r in nums_follow:
                return [i,nums_follow.index(r)+i+1]

File: 1_TwoSum.cpp
/*Given nums = [2, 7, 11, 15], target = 9,
Because nums[0] + nums[1] = 2 + 7 = 9,
return [0, 1].*/

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>m;
        vector<int>res;
        for(int i=0;i<nums.size();i++){
            m[nums[i]]=i;
        }
        for(int i=0;i<nums.size();i++){
            int t=target-nums[i];
            if(m.count(t)&&m[t]!=i){
                res.push_back(i);
                res.push_back(m[t]);
                break;
            }
        }
        return res;
    }
};
